packages/iov-tendermint-rpc/src/rpcclients/websocketclient.ts

```diff
@@ -200,7 +195,15 @@ export class WebsocketClient implements RpcStreamingClient {
    * so this should be required for testing purposes only.
    */
   public async connected(): Promise<void> {
-    return this.socket.connected;
+    return new Promise(resolve => {
```

Comment:
`await this.socket.connectionStatus.waitFor(ConnectionStatus.Connected)`

This does what you implement below and also resolves when the value is `ConnectionStatus.Connected` already.